drivers/accel/adxl355/adxl355.c

```diff
@@ -0,0 +1,814 @@
+		status = -1;
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Reads the raw output data of each axis and converts it to g.
+ *
+ * @param dev     - The device structure.
+ * @param x       - X-axis's output data.
+ * @param y       - Y-axis's output data.
+ * @param z       - Z-axis's output data.
+ *
+ * @return status - Result of the reading procedure.
+*******************************************************************************/
+int adxl355_get_xyz(struct adxl355_dev *dev, float *x, float *y, float *z)
+{
+	uint8_t array_raw_x[GET_ADXL355_TRANSF_LEN(ADXL355_XDATA)] = {0};
+	uint8_t array_raw_y[GET_ADXL355_TRANSF_LEN(ADXL355_YDATA)] = {0};
+	uint8_t array_raw_z[GET_ADXL355_TRANSF_LEN(ADXL355_ZDATA)] = {0};
+	int status = 0;
+
+	status = adxl355_read_device_data(dev,ADXL355_ADDR(ADXL355_XDATA),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_XDATA),array_raw_x);
+	if (status != 0)
+		return -1;
+	*x = adxl355_accel_conversion(dev, array_raw_x);
+
+	status = adxl355_read_device_data(dev,ADXL355_ADDR(ADXL355_YDATA),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_YDATA),array_raw_y);
+	if (status != 0)
+		return -1;
+	*y = adxl355_accel_conversion(dev, array_raw_y);
+
+	status = adxl355_read_device_data(dev,ADXL355_ADDR(ADXL355_ZDATA),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_ZDATA), array_raw_z);
+	if (status != 0)
+		return -1;
+	*z = adxl355_accel_conversion(dev, array_raw_z);
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Reads the raw temperature data and converts it to degrees Celsius.
+ *
+ * @param dev      - The device structure.
+ * @param temp     - Temperature output data.
+ *
+ * @return status  - Result of the reading procedure.
+*******************************************************************************/
+int adxl355_get_temp(struct adxl355_dev *dev,
+		     float *temp)
+{
+	uint8_t array_raw_temp[2] = {0, 0};
+	int status = 0;
+
+	status = adxl355_read_device_data(dev, ADXL355_ADDR(ADXL355_TEMP),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_TEMP),array_raw_temp);
+	if (status == 0)
+		*temp = adxl355_temp_conversion(array_raw_temp);
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Reads the status register value.
+ *
+ * @param dev          - The device structure.
+ * @param status_flags - Register value.
+ *
+ * @return status   - Result of the reading procedure.
+*******************************************************************************/
+int adxl355_get_status_register(struct adxl355_dev *dev,
+				struct adxl355_status_register_flags *status_flags)
+{
+	int status = 0;
+	uint8_t reg_value = 0;
+
+	status = adxl355_read_device_data(dev, ADXL355_ADDR(ADXL355_STATUS),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_STATUS), &reg_value);
+
+	if (status == 0) {
+		status_flags->NVM_BUSY  = (reg_value >> 4) & 1;
```

Comment:
If you're using bitfields, why not pack the flag structure and an uint8_t in a union and assign the reg_value directly instead of assigning each member sequentially

Reply:
I modified the code implementation such that it uses bit fields now.
I made these changes on the assumption that the compiler will define bitfields from the least significant bit to the most significant bit.
@buha , could you offer your input here?

Reply:
Starting with C51 Version 5 and later the order of allocation of bit-fields within a unit is LSB to MSB.